### ISSUE TITLE
TPUs available to try

### ISSUE THREAD

**Issue by brianprichardson** [glinscott/leela-chess#60]
When things look appropriate, of course:

http://www.theregister.co.uk/2018/02/12/google_cloud_tpu/

**Comment by grondilu** [comment]
Didn't Deepmind use like thousands of TPUs during several hours?  At 6.5$/TPU/h, that means that renting any comparable amount of processing would cost several thousands of dollars.

Maybe it's better to focus on distributed GPUs.

**Comment by kb1900** [comment]
doing some quick bath at $6.5 for 2tpus puts the cost of 4 hours of training of 5k tpus around 30k dollars. i believe thats what deepmind used

**Comment by brianprichardson** [comment]
There are different phases for things like generating test games (5,000 TPUs), training (64 TPUs), and evaluating and playing.  This is from the AGZ paper:

Mastering Chess and Shogi by Self-Play with a
General Reinforcement Learning Algorithm

"We applied the AlphaZero algorithm to chess, shogi, and also Go. Unless otherwise specified,
the same algorithm settings, network architecture, and hyper-parameters were used for all
three games. We trained a separate instance of AlphaZero for each game. Training proceeded
for 700,000 steps (mini-batches of size 4,096) starting from randomly initialised parameters,
using 5,000 first-generation TPUs (15) to generate self-play games and 64 second-generation
TPUs to train the neural networks.1 Further details of the training procedure are provided in the
Methods."

It is difficult for me to keep them straight (first gen TPUs v 2nd), and several different AlphaGos have been described, along with different hyperparameter training metrics.  In any case, I just thought I would mention the TPU option.
